app/controllers/application_controller.rb

```diff
@@ -36,9 +36,16 @@ def update_ldap_attributes(user)
     user.ldap_group_string = ldap_attributes['WEBAUTH_LDAPPRIVGROUP']
     user.sucard_number = ldap_attributes['WEBAUTH_LDAP_SUCARDNUMBER']
     user.affiliation = ldap_attributes['WEBAUTH_LDAP_SUAFFILIATION']
+    user.ldap_email = ldap_email
     user.save if user.changed?
   end
 
+  def ldap_email
+    return ldap_attributes['WEBAUTH_EMAIL'] unless ldap_attributes['WEBAUTH_EMAIL'].nil?
+    "#{user_id}@stanford.edu" if ldap_attributes['WEBAUTH_LDAP_SUEMAILSTATUS'] == 'active'
```

Comment:
Do you mean to `return` here?  I think it will always fall to `nil` in this case.